plugins/base/src/test/kotlin/model/ModelUtils.kt

```diff
@@ -0,0 +1,50 @@
+package model
+
+import org.jetbrains.dokka.model.Documentable
+import org.jetbrains.dokka.model.Module
+import org.jetbrains.dokka.model.doc.DocumentationNode
+import testApi.testRunner.AbstractCoreTest
+
+abstract class AbstractModelTest : AbstractCoreTest() {
+
+    fun inlineModelTest(
+        query: String,
+        platform: String = "jvm",
+        targetList: List<String> = listOf("jvm"),
+        block: Module.() -> Unit
+    ) {
+        val configuration = dokkaConfiguration {
+            passes {
+                pass {
+                    sourceRoots = listOf("src/")
+                    analysisPlatform = platform
+                    targets = targetList
+                }
+            }
+        }
+
+        testInline(query.trimIndent(), configuration) {
+            documentablesTransformationStage = block
+        }
+    }
+
+    operator fun List<Documentable?>?.div(name: String): Documentable? = this?.let { d -> d.find { it?.name == name } }
+
+    operator fun Documentable?.div(name: String): Documentable? = this?.children / name
+    fun <T> T?.assertNotNull(name: String): T = this.also { assert(this != null) { "$name should not be null" } }
+        .let { it!! }
+
+    fun <T> Any?.assertIsInstance(name: String): T =
+        this.let { it as? T }.also { assert(this != null) { "$name should not be null" } }.let { it!! }
```

Comment:
This is not working as you think it is. Parameter `T` needs to be reified for cast `it as? T` to work and IDE is telling you that!

Reply:
```kt
inline fun <T> Any?.assertIsInstance(name: String): T = 
    this as T ?: throw AssertionError("$name is not ${T::class.name}")
```
is way simpler and thas what you think it does.